staging/src/k8s.io/apimachinery/pkg/apimachinery/announced/group_factory.go

```diff
@@ -89,6 +89,17 @@ func (gmf *GroupMetaFactory) Announce(groupFactoryRegistry APIGroupFactoryRegist
 		panic(err)
 	}
 	return gmf
+
+}
+
+// AnnounceIfNotAnnounced is the same as Announce, except that it returns nil
```

Comment:
I see. I'm trying to think of a way to avoid this. Maybe we can keep a sync.Once variable near where the three things are declared, like `var APITypeInstall sync.Once` and then client/server/test code can use that to wrap their install?